tests/transforms/test_transform_can_be_hoisted.py

```diff
@@ -0,0 +1,92 @@
+from xdsl.builder import Builder, ImplicitBuilder
+from xdsl.dialects import builtin, func, scf, test
+from xdsl.dialects.arith import Addf, Constant
+from xdsl.dialects.builtin import (
+    FloatAttr,
+    FunctionType,
+    IndexType,
+    ModuleOp,
+    f32,
+    i32,
+)
+from xdsl.dialects.scf import For, Yield
+from xdsl.ir import Block, BlockArgument, Region
+from xdsl.transforms.loop_invariant_code_motion import can_be_hoisted
+
+index = IndexType()
+
+
+def test_for_with_loop_invariant_verify():
+    """Test for with loop-variant variables"""
+
+    lower = Constant.from_int_and_width(0, IndexType())
+    upper = Constant.from_int_and_width(42, IndexType())
+    step = Constant.from_int_and_width(3, IndexType())
+    carried = Constant.from_int_and_width(1, IndexType())
+
+    a = test.TestOp(result_types=[i32])
+    b = test.TestOp(result_types=[i32])
+    block = Block(
+        arg_types=(
+            index,
+            index,
+        )
+    )
+    region = Region(block)
+    with ImplicitBuilder(block) as (arg0, _):
+        op1 = test.TestOp(result_types=[i32])
+        op2 = test.TestOp(result_types=[i32])
+        op3 = test.TestOp([op1, op2, arg0], result_types=[i32])
+        _f = Yield(op3)
+    for_op = For(lower, upper, step, [carried], region)
+    block0 = Block([a, b, for_op])
+    _region_outer = Region(block0)
+
+    for region in for_op.regions:
+        for op in region.block.walk():
+            if not isinstance(op, scf.Yield):
+                assert can_be_hoisted(op3, region)
```

Comment:
There's just a single region, and we have all the ops in scope, so we can just test each of the ops directly
```suggestion
    assert can_be_hoisted(op1, region)
    assert can_be_hoisted(op2, region)
    assert can_be_hoisted(op3, region)
    assert not can_be_hoisted(f, region)
```
This code currently does not test that calling the op with yield returns False